src/ray/raylet/node_manager.cc

```diff
@@ -768,6 +736,8 @@ void NodeManager::ProcessClientMessage(
           RAY_CHECK_OK(
               client->WriteMessage(static_cast<int64_t>(protocol::MessageType::WaitReply),
                                    fbb.GetSize(), fbb.GetBufferPointer()));
+          // The client is unblocked now because the wait call has returned.
+          HandleClientUnblocked(client);
```

Comment:
Should this be called only when `required_object_ids` is not empty?

Reply:
That's probably preferable, but I think currently it will just return immediately if `HandleClientBlocked` was never called. This is similar to what happens in the `ray.get` code path where `HandleClientUnblocked` can be called even if `HandleClientBlocked` was never called.

Reply:
Hm, you're right, making that change seems to fix the Java test failure. I'm not sure I fully understand it.